src/PathFinderUtils.js

```diff
@@ -106,76 +106,94 @@ function createXPUB(
   return xpub;
 }
 
-function findPath(params, onUpdate, onDone, onError) {
+var initialize = function(coin, account, segwit) {
+  var purpose = segwit ? "49'/" : "44'/";
+  var prevPath = purpose + coin + "'";
+  return new Promise((resolve, reject) => {
+    function finalize(fingerprint) {
+      return new Promise((resolve, reject) => {
+        var path = prevPath + "/" + account + "'";
+        Dongle.btc.getWalletPublicKey_async(path).then((nodeData, error) => {
+          var publicKey = compressPublicKey(nodeData.publicKey);
+          //console.log("puikeyu", publicKey);
+          var childnum = (0x80000000 | account) >>> 0;
+          //console.log("childnum", childnum);
+          var xpub = createXPUB(
+            3,
+            fingerprint,
+            childnum,
+            nodeData.chainCode,
+            publicKey,
+            Networks[coin].xpub
+          );
+          var xpub58 = encodeBase58Check(xpub);
+          resolve(xpub58);
+        });
+      });
+    }
+    Dongle.btc.getWalletPublicKey_async(prevPath).then((nodeData, error) => {
+      var publicKey = compressPublicKey(nodeData.publicKey);
+      publicKey = parseHexString(publicKey);
+      var result = bitcoin.crypto.sha256(publicKey);
+      var result = bitcoin.crypto.ripemd160(result);
+      var fingerprint =
+        ((result[0] << 24) |
+          (result[1] << 16) |
+          (result[2] << 8) |
+          result[3]) >>>
+        0;
+      resolve(finalize(fingerprint));
+    });
+  });
+};
+
+export var findPath = (params, onUpdate, onDone, onError) => {
   if (typeof Worker !== "undefined") {
     var derivationWorker = new Worker("./workers/DerivationWorker.js");
   } else {
     onError("You need to use Google Chrome");
   }
   var running = true;
-  var prevPath = "44'/" + params.coin + "'";
   var hdnode = {};
-  var initialize = function() {
-    return new Promise((resolve, reject) => {
-      function finalize(fingerprint) {
-        return new Promise((resolve, reject) => {
-          var path = prevPath + "/" + params.account + "'";
-          Dongle.btc.getWalletPublicKey_async(path).then((nodeData, error) => {
-            var publicKey = compressPublicKey(nodeData.publicKey);
-            var childnum = (0x80000000 | parseInt(params.account)) >>> 0;
-            var xpub = createXPUB(
-              3,
-              fingerprint,
-              childnum,
-              nodeData.chainCode,
-              publicKey,
-              Networks[0].xpub
-            );
-            var xpub58 = encodeBase58Check(xpub);
-            resolve(xpub58);
-          });
+
+  Dongle.init()
+    .then(comm => {
+      return Dongle.setCoinVersion(
+        comm,
+        Networks[parseInt(params.coin, 10)]
+      ).then(res => {
```

Comment:
you should not chain `.then` in multiple callbacks.
instead of
```js
foo().then(a => 
  bar(a).then( b => 
    ...
  )
)
```

it is preferable to do

```js
foo().then(a =>
  bar(a)
).then(b =>
  ...
)
```

if you still need a AND b variables in the `...` , I recommand to switch to async/await style